gensim/models/phrases.py

```diff
@@ -118,7 +118,7 @@ def _is_single(obj):
     if isinstance(peek, string_types):
         # It's a document, return the iterator
         return True, obj_iter
-    if temp_iter == obj:
+    if temp_iter is obj:
```

Comment:
Need to add a clear test for the case when this doesn't work correctly without this fix (and now it works).
Also, return mode of file please (755 -> 644).

Reply:
Fixed.